crates/biome_service/src/configuration.rs

```diff
@@ -272,10 +273,32 @@ pub fn load_editorconfig(
     {
         let AutoSearchResult {
             content,
-            file_path: _path,
+            file_path: path,
         } = auto_search_result;
         let editorconfig = biome_configuration::editorconfig::parse_str(&content)?;
-        Ok(editorconfig.to_biome())
+        let config = editorconfig.to_biome();
+
+        // test the patterns to see if they are parsable so we can emit a better diagnostic
+        if let Some(config) = &config.0 {
+            if let Some(overrides) = &config.overrides {
```

Comment:
You can use `config.0.as_ref().and_then(|c| c.overrides)` to reduce the `Some` to one, and avoid too many indentations.